lib/ansible/modules/crypto/openssl_privatekey.py

```diff
@@ -285,9 +287,13 @@ def generate(self, module):
                             self.mode = AnsibleModule._symbolic_mode_to_octal(st, self.mode)
                         except ValueError as e:
                             module.fail_json(msg="%s" % to_native(e), exception=traceback.format_exc())
-                os.chmod(self.path, self.mode)
                 privatekey_file = os.open(self.path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
-                os.write(privatekey_file, privatekey_data)
+                os.chmod(self.path, self.mode)
+                if self.cipher and self.passphrase:
```

Comment:
...this line and the next four should be gone. These lines are from the old version.

Reply:
I have removed old five lines from generate method as you pointed out.